pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/TransformContext.java

```diff
@@ -60,13 +60,12 @@ public TransformContext(Context context, Object value) {
       this.keySchema = kvSchema.getKeySchema();
       this.keyObject =
           this.keySchema.getSchemaInfo().getType() == SchemaType.AVRO
-              ? ((org.apache.pulsar.client.api.schema.GenericRecord) kv.getKey()).getNativeObject()
+              ? ((GenericObject) kv.getKey()).getNativeObject()
```

Comment:
a "nativeObject" cannot be a "GenericObject" it is always:
- a primitive type
- a AVRO GenericRecord
- a JSON node
- a Protobuf something ???
- a KeyValue object

but NEVER a GenericObject or a Pulsar GenericRecord

this this will never happen in production, I am pretty sure

Reply:
The GenericObject cast is on `kv.getKey()` not on the nativeObject

Reply:
oh ! thanks ! you are right